dashboard/app/models/pd/enrollment.rb

```diff
@@ -176,19 +176,23 @@ def resolve_user
   end
 
   def exit_survey_url
-    if [Pd::Workshop::COURSE_ADMIN, Pd::Workshop::COURSE_COUNSELOR].include? workshop.course
-      CDO.code_org_url "/pd-workshop-survey/counselor-admin/#{code}", CDO.default_scheme
-    elsif workshop.summer?
-      pd_new_workshop_survey_url(code, protocol: CDO.default_scheme)
-    elsif [Pd::Workshop::COURSE_CSP, Pd::Workshop::COURSE_CSD].include?(workshop.course) && workshop.workshop_starting_date > Date.new(2018, 8, 1)
-      CDO.studio_url "/pd/workshop_survey/day/#{workshop.sessions.size}?enrollmentCode=#{code}", CDO.default_scheme
-    else
-      CDO.code_org_url "/pd-workshop-survey/#{code}", CDO.default_scheme
-    end
+    return CDO.code_org_url "/pd-workshop-survey/counselor-admin/#{code}", CDO.default_scheme if
+      [Pd::Workshop::COURSE_ADMIN, Pd::Workshop::COURSE_COUNSELOR].include? workshop.course
+
+    return pd_new_workshop_survey_url(code, protocol: CDO.default_scheme) if
+      workshop.summer?
+
+    return CDO.studio_url "/pd/workshop_survey/day/#{workshop.sessions.size}?enrollmentCode=#{code}", CDO.default_scheme if
+      [Pd::Workshop::COURSE_CSP, Pd::Workshop::COURSE_CSD].include?(workshop.course) && workshop.workshop_starting_date > Date.new(2018, 8, 1)
+
+    return CDO.studio_url "/pd/workshop_survey/csf/post201/#{code}", CDO.default_scheme if
+      workshop.csf? && workshop.subject == Pd::Workshop::SUBJECT_CSF_201
+
+    CDO.code_org_url "/pd-workshop-survey/#{code}", CDO.default_scheme
   end
```

Comment:
With the refactor, I'm not sure what logic changed in this method.  For whatever logic did change, is there a test that should be updated?

Reply:
Yes, I use `bin/oneoff/diff_email_previews` to compare emails that contain exit survey links before and after the change.